### PR TITLE
AIR-129 Dockerfile and Makefile changes for image size reduction

### PR DESCRIPTION
**Pull request by preranamkhare**
## ISSUE(S):
https://platform9.atlassian.net/browse/AIR-129
AIR-129 Dockerfile and Makefile changes for image size reduction in vouch

## SUMMARY
The purpose is to reduce the overall size of DDU images.
Currently vouch container image is created using base image - python:2.7
A new base image which is built based on python-slim-buster can be used to reduce overall size of the image on the DDU
This new image is available on ECR.
This image has been used in the vouch container as a base image. Fixed pre-existing unit test case failure.
Also, changes related to py2-py3 porting fixed.
## ISSUE TYPE
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## IMPACTED FEATURES/COMPONENTS:
vouch container

#### Automated
kplane-smoke-test
https://teamcity.platform9.horse/viewLog.html?buildId=1645678&tab=buildResultsDiv&buildTypeId=Pf9project_DeccoBuilds_KplaneSmokeTest

#### Reviewers
/cc @pshanbhag @yogeshj83

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,6 @@ VOUCH_DIST = $(STAGE)/vouch-sdist.tgz
 FIRKINROOT = $(abspath $(SRCROOT)/../firkinize)
 FIRKIN_CODE = $(shell find $(FIRKINROOT)/firkinize -name '*.py') \
               $(FIRKINROOT)/setup.py
-FIRKIN_DIST = $(STAGE)/firkinize-sdist.tgz
```
Comment (pshanbhag) [inline Makefile:17]: You can drop `FIRKINROOT` and `FIRKIN_CODE` variables too.

**Comment by preranamkhare** [comment]
@pshanbhag 
Incorporated the review comment. FIRKINROOT is kept since it is being referred in the unit tests. (y2j)

**Review by pshanbhag — APPROVED** [review]

**Review by pshanbhag — APPROVED** [review]

**Comment by preranamkhare** [comment]
@pshanbhag - Incorporated review comments and fixed the unit test case failure. (pre-existing).
Please re-review and approve.

File: Makefile [hunk 2]
```diff
@@ -63,19 +52,19 @@ $(BUILD_DIR)/container-tag: $(BUILD_DIR)
...
 unit-test: $(VENV)
-	$(VENV)/bin/python $(VENV)/bin/pip install -e$(SRCROOT) -e$(FIRKINROOT) \
+	$(VENV)/bin/python $(VENV)/bin/pip install -e$(SRCROOT) \
 	-e$(VAULTROOT) nose && \
 	$(VENV)/bin/nosetests -vd .
 
-$(VENV)/bin/y2j: $(VENV)
-	$(VENV)/bin/python $(VENV)/bin/pip install -e$(FIRKINROOT)
+y2j: $(VENV)
+	$(VENV)/bin/python $(VENV)/bin/pip install -e$(SRCROOT)
```
Comment (pshanbhag) [inline Makefile:60]: Why is this used to setup y2j?

Comment (pshanbhag) [reply Makefile:60]: Actually, you are directly using the y2j script. There is no need to even set it up in that case. I would just drop all dependency on y2j in this Makefile.

`$(VENV)/bin/python ./y2j $(STAGE)/app_metadata.yaml`

**Review by pshanbhag — APPROVED** [review]

**Pull request merged**